File: python/백준/비트마스킹/11723 집합.py
import sys
input = sys.stdin.readline

M = int(input())

_set = set()

for _ in range(M):
    command = input().split()

    if command[0] == 'add':
        _set.add(int(command[1]))  # 정수로 변환하여 추가

    elif command[0] == 'remove':
        if int(command[1]) in _set:  # 정수로 변환 후 확인
            _set.remove(int(command[1]))

    elif command[0] == 'check':
        if int(command[1]) in _set:
            print(1)
        else:
            print(0)

    elif command[0] == 'toggle':
        if int(command[1]) in _set:
            _set.remove(int(command[1]))
        else:
            _set.add(int(command[1]))

    elif command[0] == 'all':
        _set = set(range(1, 21))  # 1부터 20까지의 집합 생성

    elif command[0] == 'empty':
        _set = set()  # 빈 집합 생성

File: cpp/백준/집합과 맵/11723 집합.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int M;
    cin >> M;

    int _set = 0;
    for (int i = 0; i < M; ++i) {
        string command;
        cin >> command;
        if (command == "all") {
            _set = (1 << 21) - 2;
        } else if (command == "empty") {
            _set = 0;
        } else {
            int x;
            cin >> x;
            int bit = 1 << x;
            if (command == "add") {
                _set |= bit;
            } else if (command == "remove") {
                _set &= ~bit;
            } else if (command == "check") {
                cout << ((_set & bit) ? 1 : 0) << "\n";
            } else if (command == "toggle") {
                _set ^= bit;
            }
        }
    }
    return 0;
}
